SU2_CFD/src/driver_structure.cpp

```diff
@@ -2203,133 +2188,262 @@ void CDriver::Iteration_Preprocessing() {
   
 }
 
-
 void CDriver::Interface_Preprocessing() {
-  
-  int rank = MASTER_NODE;
+
+  int rank = MASTER_NODE, nProcessor = 1;
   unsigned short donorZone, targetZone;
   unsigned short nVar, nVarTransfer;
-  
+
+  unsigned short nMarkerTarget, iMarkerTarget, nMarkerDonor, iMarkerDonor;
+
   /*--- Initialize some useful booleans ---*/
   bool fluid_donor, structural_donor;
   bool fluid_target, structural_target;
-  
-  bool matching_mesh;
-  
-  fluid_donor  = false;  structural_donor  = false;
-  fluid_target  = false;  structural_target  = false;
-  
 
-#ifdef HAVE_MPI
+  int markDonor, markTarget, Donor_check, Target_check, iMarkerInt, nMarkerInt;
+
+  #ifdef HAVE_MPI
+  int *Buffer_Recv_mark, iRank;
+
   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
-#endif
+  MPI_Comm_size(MPI_COMM_WORLD, &nProcessor);
+
+  if (rank == MASTER_NODE) 
+    Buffer_Recv_mark = new int[nProcessor];
+  #endif
+
+  if (config_container[ZONE_0]->GetFSI_Simulation() && nZone != 2 && rank == MASTER_NODE){
+    cout << "Error, cannot run the FSI solver on more than 2 zones!" << endl;
+    exit(EXIT_FAILURE);
+  }
+
+  /*--- Coupling between zones ---*/
+  // There's a limit here, the interface boundary must connect only 2 zones
+
+  /*--- Loops over all target and donor zones to find which ones are connected through an interface boundary (fsi or sliding mesh) ---*/
+  for (targetZone = 0; targetZone < nZone; targetZone++){
+		
+    for (donorZone = 0; donorZone < nZone; donorZone++){
   
-  /*--- Coupling between zones (limited to two zones at the moment) ---*/
-  for (targetZone = 0; targetZone < nZone; targetZone++) {
-    
-    /*--- Initialize target booleans ---*/
-    fluid_target  = false;  structural_target  = false;
-    
-    /*--- Set the target boolean: as of now, only Fluid-Structure Interaction considered ---*/
-    switch (config_container[targetZone]->GetKind_Solver()) {
-      case EULER : case NAVIER_STOKES: case RANS: fluid_target  = true;     break;
-      case FEM_ELASTICITY:            structural_target = true;   break;
-    }
-    
-    for (donorZone = 0; donorZone < nZone; donorZone++) {
-      /*--- Initialize donor booleans ---*/
-      fluid_donor  = false;  structural_donor  = false;
-      matching_mesh = config_container[donorZone]->GetMatchingMesh();
-      
-      /*--- Set the donor boolean: as of now, only Fluid-Structure Interaction considered ---*/
-      switch (config_container[donorZone]->GetKind_Solver()) {
-        case EULER : case NAVIER_STOKES: case RANS: fluid_donor  = true;    break;
-        case FEM_ELASTICITY:            structural_donor = true;  break;
+      if ( donorZone == targetZone ) // We're processing the same zone, so skip the following
+        continue;
+
+      nMarkerInt = (int) ( config_container[donorZone]->GetMarker_n_FSIinterface() / 2 );
```

Comment:
I agree that we should make the routine for checking interfaces more general without the FSI. I am fine with "Zone_Interface".

We might wait for a separate pull request to change this, as it will certainly affect the FSI work of @rsanfer